src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Vault/SetAzureSiteRecoveryVaultSettings.cs

```diff
@@ -29,6 +29,13 @@ public class SetAzureSiteRecoveryVaultSettings : SiteRecoveryCmdletBase
         #region Parameters
 
         /// <summary>
+        /// Gets or sets vault Object.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.Default, Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public System.Object Vault { get; set; }
```

Comment:
We have two different type of Vaults (ASRVault and ARSVault) out of which ASRVault will get deprecated some time in future.
Released cmdlet has 2 different parameter sets (one for each Vault type).
We want to get rid of different vault types and parameter names for this cmdlet and so added warnings for the existing 2 types and introduced a new common type with Object as parameter type for now - so that the same parameter name / set can be used for both the types till we deprecate the older one.
And once we deprecate the older Vault type, we will move the parameter type from Object to newly supported Vault type. In this case, customers interface doesn't change from cmdlet perspective. Moreover, our PMs will educate customers via documentation on the usage of new parameter set.
As for Object type, they can pass any value - but, in our code we are handling this to be either one of our known Vault types.Customers can pass objects returned by either Get-AzureRmSiteRecoveryVault or Get-AzureRmRecoveryServicesVault.